llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2367,9 +2368,29 @@ void computeKnownBits(const Value *V, const APInt &DemandedElts,
   }
 
   // Aligned pointers have trailing zeros - refine Known.Zero set
+  auto SameFunction = [](auto A, auto B) {
+    return A->getParent()->getParent() == B->getParent()->getParent();
+  };
   if (isa<PointerType>(V->getType())) {
     Align Alignment = V->getPointerAlignment(Q.DL);
     Known.Zero.setLowBits(Log2(Alignment));
+    for (auto *User : V->users()) {
+      if (auto *Load = dyn_cast<LoadInst>(User)) {
+        if (!Q.CxtI || !SameFunction(Load, Q.CxtI) ||
+            !isValidAssumeForContext(Load, Q.CxtI, Q.DT)) {
```

Comment:
As I mentioned earlier, only check this after ensuring that the result can be improved. 
```suggestion
          Known.countMinTrailingZeros() >= Log2(Load->getAlign()) || !isValidAssumeForContext(Load, Q.CxtI, Q.DT)) {
```
We need a lambda helper function to avoid duplicating this for stores.

Reply:
Fixed, decided to use ```getLoadStorePointerOperand``` and ```getLoadStoreAlignment```.